runtime/doc/diagnostic.txt

```diff
@@ -0,0 +1,524 @@
+                    • `function` : Function with signature (namespace, bufnr)
+                      that returns any of the above.
+
+                Parameters: ~
+                    {opts}       table Configuration table with the following
+                                 keys:
+                                 • underline: (default true) Use underline for
+                                   diagnostics
+                                 • virtual_text: (default true) Use virtual
+                                   text for diagnostics
+                                 • signs: (default true) Use signs for
+                                   diagnostics
+                                 • update_in_insert: (default false) Update
+                                   diagnostics in Insert mode (if false,
+                                   diagnostics are updated on InsertLeave)
+                                 • severity_sort: (default false) Sort
+                                   diagnostics by severity. This affects the
+                                   order in which signs and virtual text are
+                                   displayed
+                    {namespace}  number|nil Update the options for the given
+                                 namespace. When omitted, update the global
+                                 diagnostic options.
+
+disable({bufnr}, {namespace})                       *vim.diagnostic.disable()*
+                Disable diagnostics in the given buffer.
+
+                Parameters: ~
+                    {bufnr}      number|nil Buffer number. Defaults to the
+                                 current buffer.
+                    {namespace}  number|nil Only disable diagnostics for the
+                                 given namespace.
+
+enable({bufnr}, {namespace})                         *vim.diagnostic.enable()*
+                Enable diagnostics in the given buffer.
+
+                Parameters: ~
+                    {bufnr}      number|nil Buffer number. Defaults to the
+                                 current buffer.
+                    {namespace}  number|nil Only enable diagnostics for the
+                                 given namespace.
+
+get({bufnr}, {namespace})                               *vim.diagnostic.get()*
+                Get all diagnostics for the buffer.
+
+                Parameters: ~
+                    {bufnr}      number|nil Buffer number to get diagnistics
+                                 from. Use 0 for current buffer. When omitted,
+                                 retrieve diagnostics from all buffers.
+                    {namespace}  number|nil Limit diagnostics to the given
+                                 namespace.
+
+                Return: ~
+                    table A list of diagnostic items |diagnostic-structure|.
+
+                                                  *vim.diagnostic.get_count()*
+get_count({bufnr}, {severity}, {namespace})
```

Comment:
can we eliminate `get_count()` by adding a `severity` param to `get()` ?

Reply:
```lua
vim.diagnostic.get_count(0, vim.diagnostic.severity.ERROR)
```

becomes

```lua
#vim.diagnostic.get(0, nil, { severity = vim.diagnostic.severity.ERROR })
```

It's a little more verbose but not prohibitively so. Sounds good to me.